juego.c

```diff
@@ -0,0 +1,324 @@
+        printf("\nInstrucciones:\n");
+        printf("Movimientos: w=arriba, s=abajo, a=izquierda, d=derecha\n");
+        printf("Objetivo: Encontrar los %d tesoros (T) evitando trampas (X)\n", numTesoros);
+        printf("Pistas usadas se marcan con ? \n");
+        printf("Tesoros restantes: %d\n", tesorosRestantes);
+        printf("------------------------------------------\n");
+        SetConsoleTextAttribute(hConsole, 7);
+        system("pause");
+
+        while (!juegoTerminado)
+        {
+            system("cls");
+
+            SetConsoleTextAttribute(hConsole, 14);
+            printf("\n   0 1 2 3 4 5 6 7 8 9\n");
+            printf("  +-------------------+\n");
+
+            for (i = 0; i < 10; i++)
+            {
+                printf("%d |", i);
+                for (j = 0; j < 10; j++)
+                {
+                    if (i == jugadorX && j == jugadorY)
+                    {
+                        SetConsoleTextAttribute(hConsole, 9);
+                        printf("P|");
+                        SetConsoleTextAttribute(hConsole, 14);
+                    }
+                    else if (tableroVisible[i][j] == 'T')
+                    {
+                        SetConsoleTextAttribute(hConsole, 10);
+                        printf("T|");
+                        SetConsoleTextAttribute(hConsole, 14);
+                    }
+                    else if (tableroVisible[i][j] == '?')
+                    {
+                        SetConsoleTextAttribute(hConsole, 6);
+                        printf("?|");
+                        SetConsoleTextAttribute(hConsole, 14);
+                    }
+                    else if (tablero[i][j] == 'T' || tablero[i][j] == 'X' || tablero[i][j] == '?')
+                    {
+                        printf(" |");
+                    }
+                    else
+                    {
+                        printf("%c|", tablero[i][j]);
+                    }
+                }
+                printf("\n");
+                if (i < 9)
+                {
+                    printf("  |-------------------|\n");
+                }
+                else
+                {
+                    printf("  +-------------------+\n");
+                }
+            }
+            SetConsoleTextAttribute(hConsole, 7);
+            printf("\nTesoros restantes: %d\n", tesorosRestantes);
+
+            printf("\nIngresa tu movimiento: ");
+            scanf(" %c", &comando);
+
+            nuevaX = jugadorX;
+            nuevaY = jugadorY;
+
+            if (comando == 'w')
+            {
+                nuevaX = nuevaX - 1;
+            }
+            else if (comando == 's')
+            {
+                nuevaX = nuevaX + 1;
+            }
+            else if (comando == 'a')
+            {
+                nuevaY = nuevaY - 1;
+            }
+            else if (comando == 'd')
+            {
+                nuevaY = nuevaY + 1;
+            }

```

Comment:
Esto se puede hacer tambien con un switch. A mi me gusta asi pero es una cuestion de preferencia personal, ambos funcionan y no hace gran diferencia en terminos de desempeno

```suggestion
            switch(comando){
                case 'w':
                    nuevaX--;
                    break;
                case 's':
                    nuevaX++;
                    break;
                case 'a':
                    nuevaY--;
                    break;
                case 'd':
                    nuevaY++;
                    break;
            }
```